### DISCUSSION TITLE
Permission changes made on filesystem don't take effect immediately

### DISCUSSION THREAD

**Discussion by Gasol** [actions/virtual-environments#6859] [General questions]
Recently, I have noticed permission changes made on filesystem do not take effect immediately in GitHub Action. As a workaround, I have to restart all started services. I have not encountered this problem before in any other environment. I am wondering what could causing this issue in GitHub Action? Is this behavior intentional by design, or is it a bug?

I have created a [job](https://github.com/Gasol/gh-action-playground/actions/runs/3820000352/jobs/6497993665) for an example.
As you can see. The curl command returns error (exist status != 0) on the first attempt when the `--fail` flag is given. However, after restarting the services, the command is successful  on the second attempt.

- [Start services](https://github.com/Gasol/gh-action-playground/actions/runs/3820000352/jobs/6497993665#step:8:3)
  $ sudo systemctl start nginx
  $ sudo systemctl start php8.1-fpm
- [Fix permission](https://github.com/Gasol/gh-action-playground/actions/runs/3820000352/jobs/6497993665#step:9:2)
  $ sudo usermod -aG docker www-data
- [Send request to /](https://github.com/Gasol/gh-action-playground/actions/runs/3820000352/jobs/6497993665#step:10:2)
  $ curl -vf localhost || true
 ```
> GET / HTTP/1.1
> Host: localhost
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 404 Not Found
< Server: nginx/1.18.0 (Ubuntu)
< Date: Mon, 02 Jan 2023 03:10:08 GMT
< Content-Type: text/html; charset=utf-8
< Content-Length: 162
< Connection: keep-alive

```
- [Restart services](https://github.com/Gasol/gh-action-playground/actions/runs/3820000352/jobs/6497993665#step:11:2)
  $ sudo systemctl restart nginx
  $ sudo systemctl restart php8.1-fpm
- [Send request to /](https://github.com/Gasol/gh-action-playground/actions/runs/3820000352/jobs/6497993665#step:12:2)
  $ curl -vf localhost
```
> GET / HTTP/1.1
> Host: localhost
> User-Agent: curl/7.81.0
> Accept: */*
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 200 OK
< Server: nginx/1.18.0 (Ubuntu)
< Date: Mon, 02 Jan 2023 03:10:08 GMT
< Content-Type: text/html; charset=UTF-8
< Transfer-Encoding: chunked
< Connection: keep-alive
< X-Frame-Options: SAMEORIGIN
< X-XSS-Protection: 1; mode=block
< X-Content-Type-Options: nosniff
< 
{ [1016 bytes data]

* Connection #0 to host localhost left intact
Array
(
    [USER] => www-data
    [HOME] => /var/www
    [HTTP_ACCEPT] => */*
    [HTTP_USER_AGENT] => curl/7.81.0
    [HTTP_HOST] => localhost
    [REDIRECT_STATUS] => 200
    [SERVER_NAME] => _
    [SERVER_PORT] => 80
    [SERVER_ADDR] => 127.0.0.1
    [REMOTE_USER] => 
    [REMOTE_PORT] => 57058
    [REMOTE_ADDR] => 127.0.0.1
    [SERVER_SOFTWARE] => nginx/1.18.0
    [GATEWAY_INTERFACE] => CGI/1.1
    [REQUEST_SCHEME] => http
    [SERVER_PROTOCOL] => HTTP/1.1
    [DOCUMENT_ROOT] => /home/runner/work/gh-action-playground/gh-action-playground/public
    [DOCUMENT_URI] => /index.php
    [REQUEST_URI] => /
    [SCRIPT_NAME] => /index.php
    [CONTENT_LENGTH] => 
    [CONTENT_TYPE] => 
    [REQUEST_METHOD] => GET
    [QUERY_STRING] => 
    [SCRIPT_FILENAME] => /home/runner/work/gh-action-playground/gh-action-playground/public/index.php
    [PATH_INFO] => 
    [FCGI_ROLE] => RESPONDER
    [PHP_SELF] => /index.php
    [REQUEST_TIME_FLOAT] => 1672629008.2589
    [REQUEST_TIME] => 1672629008
)
```

**Comment by mikhailkoliada** [comment]
Hello! It is not a bug, nginx is not supposed to pick up any changes immediately because it is a syscall usage, which in turn, is too expensive in terms of any system load, the results are cached after all, so it is by design solution :)
